pandas/tests/plotting/test_frame.py

```diff
@@ -204,6 +204,23 @@ def test_color_and_style_arguments(self):
         with pytest.raises(ValueError):
             df.plot(color=["red", "black"], style=["k-", "r--"])
 
+    def test_color_and_marker(self):
+        # GH21003 - code sample from 2018-05-10 is equivalent to this test
+        df = DataFrame(np.random.random((7, 4)))
+        # combining a color string and a marker letter should be allowed
+        df.plot(color="green", style="d")  # d for diamond
+
+    def test_color_list_and_marker(self):
+        # GH21003 - code sample from 2020-04-23 is equivalent to this test
+        df = DataFrame(np.random.random((7, 4)))
+        color_list = ["yellow", "red", "green", "blue"]
+        ax = df.plot(color=color_list, style="d")
+        # Before this patch was introduced, the previous line of code resulted
+        # in a plot where each individual line was assigned a list of colors:
+        # ax.lines[i].get_color() == ['yellow', 'red', 'green', 'blue']
+        # which resulted in a ValueError when plt.draw() was called.
+        assert [line.get_color() for line in ax.lines] == color_list
```

Comment:
same here, also use `ax.lines.get_marker()` to verify that `marker` is set correctly.

Reply:
Good point! This will introduce lots of duplicate code between the tests `test_color_and_style_arguments`, `test_color_and_marker` and `test_color_list_and_marker`. Namely `df.plot` and the list comprehensions. I'll see how I can refactor this.

Reply:
Is it preferred to test these four combinations of `color` and `style` arguments (`test_color_and_style_arguments` has two) in four separate test methods are all in one?

Reply:
i think this will work in one test?
```python
assert [line.get_color() for line in ax.lines] == color_list
assert all(line.get_marker() == 'd' for line in ax.lines)
```

Reply:
To be honest, i would prefer to combine both of your tests,
```python
@pytest.mark.parametrize("color, output_color_list", 
[('green', ['green']), (['green', 'red'], ['green', 'red'])]
)
def test_color_and_marker(self, color, output_color_list):
     df = DataFrame(np.random.random((7, 4)))
     ax = df.plot(color=color, style="d")
     assert [line.get_color() for line in ax.lines] == output_color_list
     assert all(line.get_marker() == 'd' for line in ax.lines)
```

And regarding `test_color_and_style_arguments`, it's kinda out of the scope of your PR, and you can just leave it aside.
@joooeey  does this sound good to you?

Reply:
Thanks, I didn't know about parametrize or how to use it! I will include `test_color_and_style_arguments` too because it deals with color and style kwd handling as well and has the exact same structure. Our testing suite code should first and foremost reflect possible uses of the method and not bug chronology. I'll get in the commit later tonight.

So something like the following for all four kwd-combinations:
```
@pytest.mark.parametrize("color, marker, raises", [...])
def test_color_and_style_arguments(self, df, color, marker):
    df = DataFrame({"x": [1, 2], "y": [3, 4]})
    [...]
```

Reply:
i would suggest to have another PR to clean up the test @joooeey

as you see, there are PRs out there doing clean-ups, PR is better to be small and focused on the issue, so that it has better chance to get merged quicker

Reply:
I committed the changes you originally suggested (with the slight difference of checking all four lines as opposed to just the first). I ended up not combining anything. When writing the changes I noticed that it's clearest that way.

Reply:
thanks! @joooeey

the error is irrelevant and I think you need to rebase the master to get CI green.

Reply:
How do I rebase master?

Reply:
you could also merge to master

```
git fetch upstream
git merge upstream/master
git push
```

Reply:
https://pandas.pydata.org/pandas-docs/stable/development/contributing.html#contributing-your-changes-to-pandas

Reply:
I get this on the last command:
```
Joey@zenbook MINGW64 /c/users/joey/pandas-joooeey (plot-style-error)
$ git push
fatal: The current branch plot-style-error has no upstream branch.
To push the current branch and set the remote as upstream, use

    git push --set-upstream origin plot-style-error
```
and then:
```
Joey@zenbook MINGW64 /c/users/joey/pandas-joooeey (plot-style-error)
$ git status
On branch plot-style-error
nothing to commit, working tree clean
```

Reply:
emm, as it said, you haven't set up upstream, so you will need to configure the remote first and then push. try the hint it gives `git push --set-upstream origin plot-style-error`

also i think there are some examples in the contributing file of which link i just put in comment, could you take a look and see if you can find any inspirations?

Reply:
hmm I'll look at Git later. In a hurry now.

Anyways I did just came up with a clean idea how to merge the two tests:
```
@pytest.mark.parametrize("color", ['green', ['yellow', 'red', 'green', 'blue']])
def test_color_and_marker(self, color):
     df = DataFrame(np.random.random((7, 4)))
     ax = df.plot(color=color, style="d--")
     green_line = ax.lines[2]
     assert green_line.get_color() == 'green'
     assert green_line.get_marker() == 'd'
     assert green_line.get_linestyle() == '--'
```
with the linestyle added to emphasize how subtle the difference between this test and `test_color_and_style_arguments` is.

Reply:
i think you still need to test for all, not only the `ax.lines[2]`.

and l do not know why you think `assert all(line.get_marker() == 'd' for line in ax.lines)` is NOT cleaner than the `for` loop you use.

anyway, good luck with git and feel free to make any changes you think is the best. Will take a look at your pr once it's green

Reply:
I'm pretty lost now. I can't even `import pandas`. I think this has been ever since I switched back to the `pandas-dev` environment from another environment a few days ago:
```
ImportError: cannot import name 'integer_op_not_supported' from 'pandas._libs.tslibs.timestamps' (c:\users\joey\pandas-joooeey\pandas\_libs\tslibs\timestamps.cp37-win_amd64.pyd)
```
I've rebuilt and installed pandas (as per the [pandas-dev contributing guide](https://dev.pandas.io/docs/development/contributing.html)) but that didn't help either. At this point I have so little clue what's going on, I feel like trying more things will only make it worse.

Reply:
Can you try
```bash
python setup.py build_ext --inplace -j 4
```
from within your pandas-dev environment?

Reply:
https://dev.pandas.io/docs/development/contributing.html#creating-a-development-environment

check this part out, wish it could help.

Reply:
@charlesdong1991  yes, that's what I've been looking at all this time. I did this a couple weeks ago. Today I've run in Git Bash:
```
git push --set-upstream origin plot-style-error
```
and in Anaconda Prompt:
```
(pandas-dev) c:\Users\Joey\pandas-joooeey>python setup.py build_ext --inplace -j 4
(pandas-dev) c:\Users\Joey\pandas-joooeey>python -m pip install -e . --no-build-isolation --no-use-pep517
(pandas-dev) c:\Users\Joey\pandas-joooeey>python setup.py build_ext --inplace -j 4
```
These appeared to run successfully.

What does that `cannot import name` error (see 1-hour-old comment) point to? How broken is my env? Do I need to re-build the entire env from scratch?? I remember that was lots of work because the commands given in the contributing guide don't work step-by-step the way it's written there. I had to look up lots of stuff. This is on Windows 10 with Anaconda.

Reply:
emm, usually a rebase and re-built and install should work then.

Where did you run get this error? pytest?

Reply:
> I remember that was lots of work because the commands given in the contributing guide don't work step-by-step the way it's written there

btw, would be nice to create a new issue if you are getting stuck at some point while following the guide.

Reply:
As I said, `import pandas` results in this error:
```
import pandas
Traceback (most recent call last):

  File "<ipython-input-1-38d4b0363d82>", line 1, in <module>
    import pandas

  File "C:\Users\Joey\pandas-joooeey\pandas\__init__.py", line 52, in <module>
    from pandas.core.api import (

  File "C:\Users\Joey\pandas-joooeey\pandas\core\api.py", line 15, in <module>
    from pandas.core.arrays import Categorical

  File "C:\Users\Joey\pandas-joooeey\pandas\core\arrays\__init__.py", line 8, in <module>
    from pandas.core.arrays.datetimes import DatetimeArray

  File "C:\Users\Joey\pandas-joooeey\pandas\core\arrays\datetimes.py", line 46, in <module>
    from pandas.core.arrays import datetimelike as dtl

  File "C:\Users\Joey\pandas-joooeey\pandas\core\arrays\datetimelike.py", line 11, in <module>
    from pandas._libs.tslibs.timestamps import (

ImportError: cannot import name 'integer_op_not_supported' from 'pandas._libs.tslibs.timestamps' (C:\Users\Joey\pandas-joooeey\pandas\_libs\tslibs\timestamps.cp37-win_amd64.pyd)
```

Reply:
so it's jupyter notebook, have you tried to shut down the notebook and restart? and then import?

Reply:
No, this is the IPython Console in Spyder. Restarting Spyder didn't help.

Reply:
`pytest pandas` fails too with `Windows fatal exception: access violation`:
```
[...]
pandas\tests\arrays\test_numpy.py ...................................................                                                       [ 15%]
pandas\tests\arrays\test_period.py ..................................................Windows fatal exception: access violation

Thread 0x00001f80 (most recent call first):
  File "C:\ProgramData\Anaconda3\envs\pandas-dev\lib\multiprocessing\pool.py", line 576 in _handle_results
  File "C:\ProgramData\Anaconda3\envs\pandas-dev\lib\threading.py", line 870 in run
  File "C:\ProgramData\Anaconda3\envs\pandas-dev\lib\threading.py", line 932 in _bootstrap_inner
  File "C:\ProgramData\Anaconda3\envs\pandas-dev\lib\threading.py", line 890 in _bootstrap

Thread 0x00004e4c (most recent call first):
  File "C:\ProgramData\Anaconda3\envs\pandas-dev\lib\multiprocessing\pool.py", line 528 in _handle_tasks
  File "C:\ProgramData\Anaconda3\envs\pandas-dev\lib\threading.py", line 870 in run
  File "C:\ProgramData\Anaconda3\envs\pandas-dev\lib\threading.py", line 932 in _bootstrap_inner
  File "C:\ProgramData\Anaconda3\envs\pandas-dev\lib\threading.py", line 890 in _bootstrap

Thread 0x00001fe8 (most recent call first):
  File "C:\ProgramData\Anaconda3\envs\pandas-dev\lib\multiprocessing\connection.py", line 810 in _exhaustive_wait
  File "C:\ProgramData\Anaconda3\envs\pandas-dev\lib\multiprocessing\connection.py", line 878 in wait
  File "C:\ProgramData\Anaconda3\envs\pandas-dev\lib\multiprocessing\pool.py", line 499 in _wait_for_updates
  File "C:\ProgramData\Anaconda3\envs\pandas-dev\lib\multiprocessing\pool.py", line 519 in _handle_workers
  File "C:\ProgramData\Anaconda3\envs\pandas-dev\lib\threading.py", line 870 in run
  File "C:\ProgramData\Anaconda3\envs\pandas-dev\lib\threading.py", line 932 in _bootstrap_inner
  File "C:\ProgramData\Anaconda3\envs\pandas-dev\lib\threading.py", line 890 in _bootstrap

Thread 0x000028a8 (most recent call first):
  File "C:\ProgramData\Anaconda3\envs\pandas-dev\lib\multiprocessing\pool.py", line 114 in worker
  File "C:\ProgramData\Anaconda3\envs\pandas-dev\lib\threading.py", line 870 in run
  File "C:\ProgramData\Anaconda3\envs\pandas-dev\lib\threading.py", line 932 in _bootstrap_inner
  File "C:\ProgramData\Anaconda3\envs\pandas-dev\lib\threading.py", line 890 in _bootstrap

Thread 0x000047c0 (most recent call first):
  File "C:\ProgramData\Anaconda3\envs\pandas-dev\lib\multiprocessing\pool.py", line 114 in worker
  File "C:\ProgramData\Anaconda3\envs\pandas-dev\lib\threading.py", line 870 in run
  File "C:\ProgramData\Anaconda3\envs\pandas-dev\lib\threading.py", line 932 in _bootstrap_inner
  File "C:\ProgramData\Anaconda3\envs\pandas-dev\lib\threading.py", line 890 in _bootstrap

Thread 0x00004cb8 (most recent call first):
  File "C:\ProgramData\Anaconda3\envs\pandas-dev\lib\multiprocessing\pool.py", line 114 in worker
  File "C:\ProgramData\Anaconda3\envs\pandas-dev\lib\threading.py", line 870 in run
  File "C:\ProgramData\Anaconda3\envs\pandas-dev\lib\threading.py", line 932 in _bootstrap_inner
  File "C:\ProgramData\Anaconda3\envs\pandas-dev\lib\threading.py", line 890 in _bootstrap

Thread 0x0000559c (most recent call first):
  File "C:\ProgramData\Anaconda3\envs\pandas-dev\lib\multiprocessing\pool.py", line 114 in worker
  File "C:\ProgramData\Anaconda3\envs\pandas-dev\lib\threading.py", line 870 in run
  File "C:\ProgramData\Anaconda3\envs\pandas-dev\lib\threading.py", line 932 in _bootstrap_inner
  File "C:\ProgramData\Anaconda3\envs\pandas-dev\lib\threading.py", line 890 in _bootstrap

Current thread 0x000014a0 (most recent call first):
  File "C:\Users\Joey\pandas-joooeey\pandas\tests\arrays\test_period.py", line 413 in test_arrow_table_roundtrip
  File "C:\ProgramData\Anaconda3\envs\pandas-dev\lib\site-packages\_pytest\python.py", line 184 in pytest_pyfunc_call
  File "C:\ProgramData\Anaconda3\envs\pandas-dev\lib\site-packages\pluggy\callers.py", line 187 in _multicall
  File "C:\ProgramData\Anaconda3\envs\pandas-dev\lib\site-packages\pluggy\manager.py", line 84 in <lambda>
  File "C:\ProgramData\Anaconda3\envs\pandas-dev\lib\site-packages\pluggy\manager.py", line 93 in _hookexec
  File "C:\ProgramData\Anaconda3\envs\pandas-dev\lib\site-packages\pluggy\hooks.py", line 286 in __call__
  File "C:\ProgramData\Anaconda3\envs\pandas-dev\lib\site-packages\_pytest\python.py", line 1479 in runtest
  File "C:\ProgramData\Anaconda3\envs\pandas-dev\lib\site-packages\_pytest\runner.py", line 135 in pytest_runtest_call
  File "C:\ProgramData\Anaconda3\envs\pandas-dev\lib\site-packages\pluggy\callers.py", line 187 in _multicall
  File "C:\ProgramData\Anaconda3\envs\pandas-dev\lib\site-packages\pluggy\manager.py", line 84 in <lambda>
  File "C:\ProgramData\Anaconda3\envs\pandas-dev\lib\site-packages\pluggy\manager.py", line 93 in _hookexec
  File "C:\ProgramData\Anaconda3\envs\pandas-dev\lib\site-packages\pluggy\hooks.py", line 286 in __call__
  File "C:\ProgramData\Anaconda3\envs\pandas-dev\lib\site-packages\_pytest\runner.py", line 217 in <lambda>
  File "C:\ProgramData\Anaconda3\envs\pandas-dev\lib\site-packages\_pytest\runner.py", line 244 in from_call
  File "C:\ProgramData\Anaconda3\envs\pandas-dev\lib\site-packages\_pytest\runner.py", line 216 in call_runtest_hook
  File "C:\ProgramData\Anaconda3\envs\pandas-dev\lib\site-packages\_pytest\runner.py", line 186 in call_and_report
  File "C:\ProgramData\Anaconda3\envs\pandas-dev\lib\site-packages\_pytest\runner.py", line 100 in runtestprotocol
  File "C:\ProgramData\Anaconda3\envs\pandas-dev\lib\site-packages\_pytest\runner.py", line 85 in pytest_runtest_protocol
  File "C:\ProgramData\Anaconda3\envs\pandas-dev\lib\site-packages\pluggy\callers.py", line 187 in _multicall
  File "C:\ProgramData\Anaconda3\envs\pandas-dev\lib\site-packages\pluggy\manager.py", line 84 in <lambda>
  File "C:\ProgramData\Anaconda3\envs\pandas-dev\lib\site-packages\pluggy\manager.py", line 93 in _hookexec
  File "C:\ProgramData\Anaconda3\envs\pandas-dev\lib\site-packages\pluggy\hooks.py", line 286 in __call__
  File "C:\ProgramData\Anaconda3\envs\pandas-dev\lib\site-packages\_pytest\main.py", line 272 in pytest_runtestloop
  File "C:\ProgramData\Anaconda3\envs\pandas-dev\lib\site-packages\pluggy\callers.py", line 187 in _multicall
  File "C:\ProgramData\Anaconda3\envs\pandas-dev\lib\site-packages\pluggy\manager.py", line 84 in <lambda>
  File "C:\ProgramData\Anaconda3\envs\pandas-dev\lib\site-packages\pluggy\manager.py", line 93 in _hookexec
  File "C:\ProgramData\Anaconda3\envs\pandas-dev\lib\site-packages\pluggy\hooks.py", line 286 in __call__
  File "C:\ProgramData\Anaconda3\envs\pandas-dev\lib\site-packages\_pytest\main.py", line 247 in _main
  File "C:\ProgramData\Anaconda3\envs\pandas-dev\lib\site-packages\_pytest\main.py", line 191 in wrap_session
  File "C:\ProgramData\Anaconda3\envs\pandas-dev\lib\site-packages\_pytest\main.py", line 240 in pytest_cmdline_main
  File "C:\ProgramData\Anaconda3\envs\pandas-dev\lib\site-packages\pluggy\callers.py", line 187 in _multicall
  File "C:\ProgramData\Anaconda3\envs\pandas-dev\lib\site-packages\pluggy\manager.py", line 84 in <lambda>
  File "C:\ProgramData\Anaconda3\envs\pandas-dev\lib\site-packages\pluggy\manager.py", line 93 in _hookexec
  File "C:\ProgramData\Anaconda3\envs\pandas-dev\lib\site-packages\pluggy\hooks.py", line 286 in __call__
  File "C:\ProgramData\Anaconda3\envs\pandas-dev\lib\site-packages\_pytest\config\__init__.py", line 124 in main
  File "C:\ProgramData\Anaconda3\envs\pandas-dev\Scripts\pytest-script.py", line 10 in <module>

(pandas-dev) c:\Users\Joey\pandas-joooeey>
```

Reply:
emm, have no idea, i really think 
```
python setup.py build_ext --inplace -j 4
python -m pip install -e . --no-build-isolation --no-use-pep517
```
would solve the issue

@WillAyd could you help with the environment issue?